src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsUrlTest.java

```diff
@@ -0,0 +1,200 @@
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.internal;
+
+import static com.google.common.truth.Truth.assertWithMessage;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+
+import javax.xml.parsers.SAXParser;
+import javax.xml.parsers.SAXParserFactory;
+
+import org.junit.jupiter.api.Test;
+import org.xml.sax.Attributes;
+import org.xml.sax.helpers.DefaultHandler;
+
+import com.puppycrawl.tools.checkstyle.internal.utils.CheckUtil;
+
+public class XdocsUrlTest {
+
+    public static final String PACKAGE_NAME =
+            "src/main/java/com/puppycrawl/tools/checkstyle/checks";
+
+    public static final String SUFFIX_CHECK = "Check";
+
+    public static final String CHECKS = "checks";
+
+    public static final String CHECKSTYLE = "checkstyle";
+
+    public static final String FILTERS = "filters";
+
+    public static final String FILE_FILTERS = "filefilters";
+
+    public static final String MISC = "misc";
+
+    public static final String ANNOTATION = "annotation";
+
+    public static final String COMMENTS_INDENTATION = "CommentsIndentation";
+
+    public static final String INDENTATION = "Indentation";
+
+    public static final String SUPPRESS_WARNINGS_HOLDER = "SuppressWarningsHolder";
+
+    public static final Path AVAILABLE_CHECKS_PATH = Paths.get("src/xdocs/checks.xml");
+
+    private static Map<String, ArrayList<String>> getXdocsMap() throws IOException {
+        final Map<String, ArrayList<String>> checksNamesMap = new HashMap<>();
+        final Set<Class<?>> checkSet = CheckUtil.getCheckstyleModules();
+        for (Class<?> check : checkSet) {
+            final String[] splitArray = check.getName().split("\\.");
+            String subPackage = splitArray[splitArray.length - 2];
+            if (!subPackage.equals(CHECKSTYLE)
+                    && !subPackage.equals(FILTERS)
+                    && !subPackage.equals(FILE_FILTERS)) {
+                if (subPackage.equals(CHECKS)) {
+                    subPackage = MISC;
+                }
+                final String checkName = splitArray[splitArray.length - 1];
```

Comment:
You could just do `check.getSimpleName()`

Reply:
done